standup/status/search.py

```diff
@@ -0,0 +1,208 @@
+from django.db.models import Q
+
+
+# FIXME(willkg): This ignores the variety of other whitespace characters in unicode.
+WHITESPACE = u' \t\r\n'
+
+
+def to_tokens(text):
+    """Breaks the search text into tokens"""
+    in_quotes = False
+    escape = False
+
+    tokens = []
+    token = []
+
+    for c in text:
+        if c == u'\\':
+            escape = True
+            token.append(c)
+            continue
+
+        if in_quotes:
+            if not escape and c == u'"':
+                in_quotes = False
+            token.append(c)
+
+        elif not escape and c == u'"':
+            in_quotes = True
+            token.append(c)
+
+        elif c in WHITESPACE:
+            if token:
+                tokens.append(u''.join(token))
+                token = []
+
+        else:
+            token.append(c)
+
+        escape = False
+
+    if in_quotes:
+        # Finish off a missing quote
+        if token:
+            token.append(u'"')
+        else:
+            tokens[-1] = tokens[-1] + u'"'
+
+    if token:
+        # Add last token
+        tokens.append(u''.join(token))
+
+    return tokens
+
+
+class ParseError(Exception):
+    pass
+
+
+def unescape(text):
+    """Unescapes text
+
+    >>> unescape(u'abc')
+    u'abc'
+    >>> unescape(u'\\abc')
+    u'abc'
+    >>> unescape(u'\\\\abc')
+    u'\\abc'
+
+    """
+    # Note: We can ditch this and do it in tokenizing if tokenizing
+    # returned typed tokens rather than a list of strings.
+    new_text = []
+    escape = False
+    for c in text:
+        if not escape and c == u'\\':
+            escape = True
+            continue
+
+        new_text.append(c)
+        escape = False
+
+    return u''.join(new_text)
+
+
+def build_match(field, token):
+    return Q(**{'%s__icontains' % field: unescape(token)})
+
+
+def build_match_phrase(field, token):
+    return Q(**{'%s__icontains' % field: unescape(token)})
```

Comment:
I wasn't sure how else to build this other than with `icontains`. It means that the word has to appear exactly in the content field. It also means that it'll pick up substrings. So if you searched for "antenna", it'd pick up all standup status entries for "antenna", "antennas", "antenna-loadtest", etc. That might be fine, but something like "end" might not be.

The consequence of this is that this implementation of search will kind of suck.

Having said that, we don't have better options without doing a lot more work.